### PR TITLE
Add rocksdb_property_* functions

### PR DESCRIPTION
**Pull request by dvdplm**
Clean up tests to not leave garbage on failure.

The RocksDB API exposes internal stats through a set of functions that all take a property name (`&str`) and return either a `u64` or a `String`. Currently these features are not exposed to rust so this PR adds them.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -143,37 +143,37 @@ fn main() {
...
-        blockopts.set_block_size(524288);
-        opts.set_block_based_table_factory(blockopts);
-        opts.set_disable_auto_compactions(true);
-
-        // let filter = new_bloom_filter(10);
-        // opts.set_filter(filter);
-
-        DB::open(&opts, path).unwrap()
-    }
+//    use rocksdb::{BlockBasedOptions, DB, Options};
```
Comment (tomusdrw) [inline src/main.rs]: What was wrong with this test?

**Review by tomusdrw — APPROVED** [review]
Looks sane, I'm not an expert on FFI though.

File: src/main.rs (hunk 1 above)
Comment (dvdplm) [reply src/main.rs]: It was not called by anything and threw a compiler warning so I figured it was cruft left there for some reason – maybe to run tests on different hardware?

Maybe I should remove it completely?

Comment (ordian) [reply src/main.rs]: does putting `#[test]` attribute on top of the function help?

File: src/rocksdb.rs [hunk 2]
```diff
@@ -512,6 +512,70 @@ impl DB {
...
+        value
+    }
+
+    /// Query a RocksDB instance scoped to a column family for a "property" that returns a numeric
+    /// value. The available properties are listed
+    /// [here](https://github.com/facebook/rocksdb/blob/fefd4b98c572a09da47fb100da9e1472d28c0d08/include/rocksdb/db.h#L522)
+    /// but note that only properties returning a numeric value can be queried using this function.
+    pub fn get_int_property_cf(&self, cf: Column , prop: &str) -> u64 {
+        let string_value = self.get_property_value_cf(cf, prop);
+        string_value.parse::<u64>().unwrap_or(0)
```
Comment (ordian) [inline src/rocksdb.rs]: maybe add a comment that this is a tmp workaround for https://github.com/facebook/rocksdb/issues/5249 (or create an issue)

Comment (dvdplm) [reply src/rocksdb.rs]: I noticed today that upstream `rust-rocksdb` does this too: https://github.com/rust-rocksdb/rust-rocksdb/blob/184440b2095979baeb747ccf300022198e9606ed/src/db.rs#L1529

Working on a proper patch in facebook/rocksdb.

File: src/main.rs (hunk 1 above)
Comment (dvdplm) [reply src/main.rs]: @ordian whatever this code is I don't think it ever was a test (the fn takes arguments for instance). The tests for this crate are in `tests/`. I think this `main.rs` file is all cruft tbh, it seems like some sort of testing ground for the examples in the `README` etc...

Comment (ordian) [reply src/main.rs]: yeah, then removing all that cruft is fine

Comment (dvdplm) [reply src/main.rs]: I moved `src/main.rs` to `examples/basic_operation.rs` and removed all code therein that didn't work or I didn't understand the purpose of. Also removed the unused features from the crate.

**Review by ordian — APPROVED** [review]
Nice work!

**Comment by dvdplm** [comment]
Published as 0.5.1

**Pull request merged**